holoviews/plotting/mpl/element.py

```diff
@@ -314,16 +315,21 @@ def _set_axis_limits(self, axis, view, subplots, ranges):
             if l != r:
                 if valid_lim(l):
                     axis.set_xlim(left=l)
+                    xset = True
                 if valid_lim(r):
                     axis.set_xlim(right=r)
+                    xset = True
 
             if self.invert_yaxis or any(p.invert_yaxis for p in subplots):
                 t, b = b, t
             if b != t:
                 if valid_lim(b):
                     axis.set_ylim(bottom=b)
+                    yset = True
                 if valid_lim(t):
                     axis.set_ylim(top=t)
+                    yset = True
+        axis.autoscale_view(scalex=not xset, scaley=not yset)
```

Comment:
Seems like a reasonable change although why not invert the booleans and use the same names as matplotlib ``scalex`` and ``scaley``? That would seem clearer to me...